src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -728,7 +728,7 @@ public void visitCtDo(CtDo doLoop) {
 			printer.writeSeparator(";").writeln();
 		} else {
 			elementPrinterHelper.printList(ctEnum.getEnumValues(),
-					null, false, null, false, false, ",", true, false, ";",
+					null, false, null, false, false, "," + DefaultJavaPrettyPrinter.LINE_SEPARATOR, false, false, ";",
 					enumValue -> scan(enumValue));
```

Comment:
it is not actually allowed to send token which contains two different token types: "," and EOL

Solution:
replace
```java
enumValue -> scan(enumValue)
```
with 
```java
enumValue -> {
   printer.writeln();
   scan(enumValue)
}
```

Reply:
Great thanks!